pylint/checkers/base.py

```diff
@@ -2593,6 +2594,41 @@ def _check_type_x_is_y(self, node, left, operator, right):
         self.add_message("unidiomatic-typecheck", node=node)
 
 
+class ThreadingChecker(_BasicChecker):
+    """Checks for threading module
+
+    - useless with lock - locking used in wrong way that has no effect (with threading.Lock():)
+    """
+
+    msgs = {
+        "W9990": (  # TODO
+            "Useless usage of with statement and %s()",
+            "useless-with-lock",
+            "Used when a new lock instance is created by using with statement "
+            "which has no effect, instead of using existing instance to acquire lock. "
+            "Valid for various locking objects: Lock, RLock, Condition, Semaphore, BoundedSemaphore.",
+        ),
+    }
+
+    @utils.check_messages("useless-with-lock")
+    def visit_with(self, node: nodes.With) -> None:
+        locks = {
+            "threading.Lock",
+            "threading.RLock",
+            "threading.Condition",
+            "threading.Semaphore",
+            "threading.BoundedSemaphore",
+        }
+        pairs = node.items
+        for manager, _ in pairs:
+            if isinstance(manager, nodes.Call):
+                inf = safe_infer(manager.func)
+                if inf is not None:
+                    qname = inf.qname()
+                    if qname in locks:
+                        self.add_message("useless-with-lock", node=node, args=qname)
```

Comment:
```suggestion
        context_managers = [c for c, _ in node.items if isinstance(c, nodes.Call)
        for context_manager in context_managers:
            infered_function = safe_infer(context_manager.func)
            if infered_function is None:
                continue
            qname = infered_function .qname()
            if qname in locks:
                self.add_message("useless-with-lock", node=node, args=qname)
```
Some nitpick to reduce the indentation level.

Reply:
Updated